libibumad/man/umad_get_ca_namelist.3.md

```diff
@@ -0,0 +1,54 @@
+
+---
+date: "May 1, 2018"
+footer: "OpenIB"
+header: "OpenIB Programmer's Manual"
+layout: page
+license: 'Licensed under the OpenIB.org BSD license (FreeBSD Variant) - See COPYING.md'
+section: 3
+title: UMAD_GET_CA_NAMELIST
+---
+
+# NAME
+
+umad_get_ca_namelist - get list of available InfiniBand device names
+
+# SYNOPSIS
+
+```c
+#include <infiniband/umad.h>
+
+char **umad_get_ca_namelist(void);
```

Comment:
This should really be const char *

I also wonder if this should be a struct not a char *??

When you go to do the sysfs stuff you are really going to want a struct to hold the ifidx from discovery to opening the device.

Reply:
1. I will change to const char *. 

2. We use the function in order to call umad_get_ca() with device name (as const char *)
Hence, I'n my opinion, it is better to return the same type that we will use.

3. What do you mean by ifidx ? is it interface index ?
If so, doesn't it mean we need to make sure index is persistence between umad calls ?

Reply:
ifidx is persisted by the kernel and when we say 'use netlink' it really  means to use ifidx everywhere and never, ever the interface name. This is because we have renaming now, so referring to an interface by name is inherently racey.

This is something that does need to get fixed my preference would be to see new APIs support this working model, ie return a linked list of 'struct umad_device' that can be passed to some future new umad_get_ca() which can use the ifidx to open the /dev/ file. For now callers can use some 'umad_device->name' to pass to umad_get_ca()

Linked list is because it is ABI and structure sizes cannot change